packages/reactive-dict/reactive-dict.js

```diff
@@ -1,117 +1,114 @@
 // XXX come up with a serialization method which canonicalizes object key
 // order, which would allow us to use objects as values for equals.
-var stringify = function (value) {
-  if (value === undefined)
+function stringify(value) {
+  if (value === undefined) {
     return 'undefined';
+  }
   return EJSON.stringify(value);
-};
-var parse = function (serialized) {
-  if (serialized === undefined || serialized === 'undefined')
+}
+
+function parse(serialized) {
+  if (serialized === undefined || serialized === 'undefined') {
     return undefined;
+  }
   return EJSON.parse(serialized);
-};
+}
 
-var changed = function (v) {
+function changed(v) {
   v && v.changed();
-};
+}
 
 // XXX COMPAT WITH 0.9.1 : accept migrationData instead of dictName
-ReactiveDict = function (dictName) {
-  // this.keys: key -> value
-  if (dictName) {
-    if (typeof dictName === 'string') {
-      // the normal case, argument is a string name.
-      // _registerDictForMigrate will throw an error on duplicate name.
-      ReactiveDict._registerDictForMigrate(dictName, this);
-      this.keys = ReactiveDict._loadMigratedDict(dictName) || {};
-      this.name = dictName;
-    } else if (typeof dictName === 'object') {
-      // back-compat case: dictName is actually migrationData
-      this.keys = dictName;
-    } else {
-      throw new Error("Invalid ReactiveDict argument: " + dictName);
-    }
-  } else {
-    // no name given; no migration will be performed
+export class ReactiveDict {
+  constructor(dictName, dictData = {}) {
+    // this.keys: key -> value
     this.keys = {};
-  }
 
-  this.allDeps = new Tracker.Dependency;
-  this.keyDeps = {}; // key -> Dependency
-  this.keyValueDeps = {}; // key -> Dependency
-};
+    if (dictName) {
+      // name given; migration will be performed
+      if (typeof dictName === 'string') {
+        // the normal case, argument is a string name.
+        // _registerDictForMigrate will throw an error on duplicate name.
+        ReactiveDict._registerDictForMigrate(dictName, this);
+        // Use _setObject to make sure values are stringified
+        this._setObject(ReactiveDict._loadMigratedDict(dictName) || dictData);
+        this.name = dictName;
+      } else if (typeof dictName === 'object') {
+        // back-compat case: dictName is actually migrationData
+        // Use _setObject to make sure values are stringified
+        this._setObject(dictName);
```

Comment:
I think calling `this._setObject` here makes sense, but I think there is a bug lingering here. If we initialize a `ReactiveDict` like `new ReactiveDict('name', { property: 'some value' })`, we'll hit this `_setObject` call. So far so good, but down in the `set` function (which we end up at via the `_setObject` call), we hit this block:

```js
    if (isNewValue) {
      changed(this.keyDeps[key]);
      if (this.keyValueDeps[key]) {
        changed(this.keyValueDeps[key][oldSerializedValue]);
        changed(this.keyValueDeps[key][value]);
      }
    }
```

The problem is the second line will fail, as `this.keyDeps` is `undefined`. We haven't initialized it yet in the constructor (it's initialized after the `_setObject` call). So we'll want to make sure it's initialized beforehand (and maybe add a test to catch this as well).